main.cpp

```diff
@@ -89,422 +89,222 @@ If you need to view an example, see: https://bitbucket.org/MatkatMusic/pfmcpptas
-}
 
-IntType& IntType::operator*=(const int other) 
-{
-    *intTypeHeap *= other;
-    return *this;
-}
-
-IntType& IntType::apply(std::function<IntType&(int&)> intFunc)
-{
-    if(intFunc)
+    Numeric& operator+=(const MyType other)
     {
-        return intFunc(*intTypeHeap);
+        *numericHeap += other;
+        return *this;
     }
-    return *this;
-}
 
-IntType& IntType::apply(void(*intFunc)(int&))
-{
-    if(intFunc)
+    Numeric& operator-=(const MyType other)
     {
-        intFunc(*intTypeHeap);
+        *numericHeap -= other;
+        return *this;
     }
-    return *this;
-}
-
-IntType& IntType::pow(int power)
-{
-    *intTypeHeap = powInternal(power);
-    return *this;
-}
-
-IntType& IntType::pow(const IntType& intRef)
-{
-    *intTypeHeap = powInternal(intRef);
-    return *this;   
-}
-
-IntType& IntType::pow(const DoubleType& doubleRef)
-{
-    *intTypeHeap = powInternal(static_cast<int>(doubleRef));
-    return *this;
-}
-
-int IntType::powInternal(int power)
-{
-    return static_cast<int>(std::pow(*intTypeHeap, power));
-}
-
-IntType& IntType::pow(const FloatType& floatRef)
-{
-    *intTypeHeap = (powInternal(static_cast<int>(floatRef)));
-    return *this;
-}
-
-//FloatType Implementation //////////////////////////////////////
-
-
-FloatType::FloatType(float floatValue)
-{
-    floatTypeHeap = new float(floatValue);
-}
-
-FloatType::~FloatType()
-{
-    delete floatTypeHeap;
-}
-
-FloatType& FloatType::operator+=( const float other) 
-{
-    *floatTypeHeap += other;
-    return *this;
-}
-
-FloatType& FloatType::operator-=(const float other)
-{
-    *floatTypeHeap -= other;
-    return *this;
-}
 
-FloatType& FloatType::operator/=( const float other) 
-{
-    *floatTypeHeap /= other;
-    return *this;
-}
-
-FloatType& FloatType::operator*=(const float other) 
-{
-    *floatTypeHeap *= other;
-    return *this;
-}
-
-FloatType& FloatType::apply(std::function<FloatType&(float&)> floatFunc)
-{
-    if(floatFunc)
+    Numeric& operator/=(const MyType other)
     {
-        return floatFunc(*floatTypeHeap);
+        // if your template type is an int
+        // if your parameter's type is also an int
+        //                 if your parameter is 0
+        //                         don't do the division
+        //         else if it's less than epsilon
+        //                 dont do the divison
+        // else if it's less than epsilon
+        //         warn about doing the division
+        if(std::is_same<int, MyType>::value)
+        {
+            if(std::is_same<decltype(other), int>::value)
+            {
+                if(other == 0.f)
+                {
+                    return *this;
+                }
+                else if(std::numeric_limits<MyType>::is_integer)
+                {
+                    return *this;
+                }
+                else
+                {
+                    std::cout<<"Warning division by zero";
+                }
+
+            } 
+        } 
```

Comment:
You never actually do any division...
- [ ] you have some logic issues that don't follow the plain-english code.
- [ ] you're not using epsilon
- [ ] that `other == 0.f` when both types are `int` is wrong because it forces 'other' to convert to `float`
- [ ] revise